pedal/gpt/commands.py

```diff
@@ -0,0 +1,124 @@
+import json
+import os
+
+from pedal.core.report import MAIN_REPORT
+from pedal.gpt.constants import TOOL_NAME
+from pedal.gpt.feedbacks import gpt_prompt_feedback
+
+try:
+    import openai
+except ImportError:
+    print("OpenAI library not found!")
+
+__all__ = ['gpt_get_api_key', 'gpt_run_prompts']
+
+
+def gpt_get_api_key(default):
+    """
+    Takes a fallback string and returns the API key, or the fallback
+    if the environment variable doesn't exist.
+
+    Args:
+        default (str): The fallback API key. Probably passed as a CLI arg
+    """
+    return os.getenv("OPENAI_API_KEY", default=default)
+
+def gpt_run_prompts(api_key, code=None, report=MAIN_REPORT):
+    """
+    Evaluates the prompts and attach the results to the Report.
+
+    Args:
+        api_key:
+        code (str or None): The code to evaluate with TIFA. If ``code`` is not
+            given, then it will default to the student's main file.
+        report (:class:`pedal.core.report.Report`): The Report object to
+            attach results to.
+    """
+    if not code:
+        code = report.submission.main_code
+
+    prompt = f"""{code}"""
+
+    messages = [
+        {'role': 'system',
+         'content': "You are an intelligent tutor for a introductory computer science course in Python. You never give answers but do give helpful tips guide students with their code."},
```

Comment:
The reason this is inside the function is because it relies on the student code passed to the function. We can discuss tomorrow how to move it out if feasible, I can't think of an easy and good way to do that